File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
    
        hashtb = {}
        for i, n in enumerate(nums):
            if target-n in hashtb:
                return [hashtb[target-n], i]
            hashtb[n] = i
            
    # Time Complexity: O(n)
    # Space Complexity: O(n)

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> m = new HashMap<>();
        for (int i = 0; i < nums.length; i++){
            if (m.containsKey(target - nums[i]))
                return new int[] {m.get(target-nums[i]), i};
            m.put(nums[i], i);
        }
        return null;
    }
}